ajbleakley/register-app
# Feature - Use rich domain model for user entity

### What is the (feature|problem)?

Relates to: [AS AN application entity I WANT TO benefit from a rich domain model SO THAT I can encapsulate behaviour](https://github.com/ajbleakley/register-app/issues/18)

### What is the solution?

Apply rich domain model to the application entity - see [my proposal](https://github.com/ajbleakley/register-app/issues/18) for more info.

### What areas of the app does it impact?

No impact to application clients, but should increase maintainability of application code base for developers.

### How to test?

Run all application tests to check they still pass: `composer test`
Check application still runs in Docker container: `docker compose up --build -d`
